src/Service/ApplicationFactory.php

```diff
@@ -26,6 +26,12 @@ class ApplicationFactory implements FactoryInterface
      */
     public function createService(ServiceLocatorInterface $serviceLocator)
     {
-        return new Application($serviceLocator->get('Config'), $serviceLocator);
+        return new Application(
+            $serviceLocator->get('Config'),
+            $serviceLocator,
+            $serviceLocator->get('EventManager'),
```

Comment:
Add noted in my issue summary, I needed to make this change due to a race
condition with EM injection.

Since most people never directly instantiate, this will affect very few
people. I'll be sure to call it out in the migration guide.
On Oct 1, 2015 6:42 AM, "Marco Pivetta" <notifications@github.com> wrote:

> In src/Service/ApplicationFactory.php
> <https://github.com/zendframework/zend-mvc/pull/31#discussion_r40903811>:
>
> > @@ -26,6 +26,12 @@ class ApplicationFactory implements FactoryInterface
> >       */
> >      public function createService(ServiceLocatorInterface $serviceLocator)
> >      {
> > -        return new Application($serviceLocator->get('Config'), $serviceLocator);
> > +        return new Application(
> > +            $serviceLocator->get('Config'),
> > +            $serviceLocator,
> > +            $serviceLocator->get('EventManager'),
>
> As mentioned above, it may be a better idea to keep this as it was before
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-mvc/pull/31/files#r40903811>.
>